cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageDataset.java

```diff
@@ -319,7 +320,7 @@ private void addOperationReferenceRecord(byte[] direction, EndPoint endPoint, Pr
     Map<Long, Set<ProgramRunId>> checksumsWithProgramRunsInRange
       = getChecksumsWithProgramRunsInRange(direction, endPoint, start, end);
 
-    Set<ProgramRunOperations> result = new HashSet<>();
+    Set<ProgramRunOperations> result = new LinkedHashSet<>();
```

Comment:
why is this needed

Reply:
same. To preserve the descending ordering.